### DISCUSSION TITLE
Using clang-cl and vcpkg manifest mode in VS2022 project

### DISCUSSION THREAD

**Discussion by sidy3d** [microsoft/vcpkg#26038] [Q&A] [answered]
Hi,

I have a MSBuild project in Visual Studio 2022 which uses vcpkg through manifest mode to manage dependencies. There are lots of computationally intense things in the project so I wanted to use clang-cl to try to improve performance (ex. MSVC not inclining functions in the Eigen math library). 

My understanding is that this requires two steps

- Change the project properties toolset to LLVM (clang-cl), this will compile my code using clang-cl
- Use a custom triplet so that dependencies are compiled using clang-cl

 I am having some trouble setting this up. Is there a custom triplet available for clang-cl?

Thanks,

**Comment by Neumann-A** [comment, marked as answer]
You can either look here:
https://github.com/microsoft/vcpkg/pull/25897
and look at the triplet changes or
look here:
https://github.com/Neumann-A/my-vcpkg-triplets and look at the -llvm stuff.
In the latter case be aware that I use some tricks to avoid binary caching so you probably want something simpler. 

Not all ports are currently correctly switching to clang-cl or forwarding compiler flags but https://github.com/microsoft/vcpkg/pull/25897 gives a good view what works and what doesn't

**Reply by sidy3d** [reply]
Thanks for the information. Seems much more complicated then I expected. Looks like I will have to improve my knowledge of CMake first.
